xla/service/gpu/transforms/explicit_nccl_group_async_wrapper_test.cc

```diff
@@ -0,0 +1,106 @@
+/* Copyright 2024 The OpenXLA Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "xla/service/gpu/transforms/explicit_nccl_group_async_wrapper.h"
+
+#include <memory>
+
+#include <gtest/gtest.h>
+#include "absl/status/statusor.h"
+#include "absl/strings/string_view.h"
+#include "xla/hlo/ir/hlo_instruction.h"
+#include "xla/hlo/testlib/filecheck.h"
+#include "xla/service/gpu/backend_configs.pb.h"
+#include "xla/side_effect_util.h"
+#include "xla/tests/hlo_test_base.h"
+#include "xla/tsl/lib/core/status_test_util.h"
+#include "xla/tsl/platform/statusor.h"
+
+namespace xla::gpu {
+namespace {
+
+using ExplicitNcclGroupAsyncWrapperTest = HloTestBase;
```

Comment:
We removed Nccl from all type names in XLA:GPU and we use generic "collectives" + generic collectives APIs everywhere. Can you rename it to `ExplicitCollectiveGroupAsyncWrapper`.

Reply:
Sure, but this pass might be meaningless on non-Nvidia hardware. We also are still using `_nccl_group` as the frontend attribute, do you think that needs to be changed too?

Reply:
Yes, eventually most mentions of nccl should go away (when they have a generic replacement).

Reply:
I'll change it to `_collectives_group`